### PR TITLE
UI/slot management

### PR DESCRIPTION
**Pull request by Desmondmies**
Slot info Tooltip

- [x] #55 
- [x] #20 
- [x] #21 
- [x] #22 
- [x] #23

### CODE REVIEW

File: edtLHD/src/main/java/fr/univtln/lhd/controllers/SlotInfoController.java [hunk 1]
```diff
@@ -0,0 +1,339 @@
...
+    /**
+     * Modify a slot
+     * Gets all information from each field
+     * Calls a method on parent Controller to modify old Slot with new modified information
+     */
+    private void modifySlot(){
+        Slot oldSlot = getSlotUI().getSlot();
+
+        if(Schedule.updateInSchedule(oldSlot, getSlotFromPanel()))
+            System.out.println("MODIFY ERROR");
```
Comment (anastygnome) [inline edtLHD/src/main/java/fr/univtln/lhd/controllers/SlotInfoController.java:283]: Can you please use a logger?

File: edtLHD/src/main/java/fr/univtln/lhd/controllers/SlotInfoController.java [hunk 2]
```diff
@@ -0,0 +1,339 @@
...
+    private void deleteSlot(){
+        Slot slotToDelete = getSlotFromPanel();
+        try {
+            slotToDelete.setId( getSlotUI().getSlot().getId() );
+        } catch (IdException e){
+            throw new RuntimeException();
+        }
+
+        if(Schedule.deleteInSchedule( slotToDelete ))
+            System.out.println("DELETE ERROR");
```
Comment (anastygnome) [inline edtLHD/src/main/java/fr/univtln/lhd/controllers/SlotInfoController.java:302]: Same here

File: edtLHD/src/main/java/fr/univtln/lhd/controllers/SlotInfoController.java [hunk 3]
```diff
@@ -0,0 +1,339 @@
...
+     * Delete a slot
+     * Gets all information from each field
+     * Calls a method on parent Controller to delete the selected slot
+     */
+    private void deleteSlot(){
+        Slot slotToDelete = getSlotFromPanel();
+        try {
+            slotToDelete.setId( getSlotUI().getSlot().getId() );
+        } catch (IdException e){
+            throw new RuntimeException();
```
Comment (anastygnome) [inline edtLHD/src/main/java/fr/univtln/lhd/controllers/SlotInfoController.java:298]: Would an IllegalStateException make the sonar smell disappear?

**Review by anastygnome — CHANGES_REQUESTED** [review]
Amazing work, if you could just get rid of the System.err calls :)

**Review by Theo-Hafsaoui — APPROVED** [review]
Great work indeed

**Review by anastygnome — APPROVED** [review]

**Pull request merged**